### ISSUE TITLE
sunsim test failures

### ISSUE THREAD

**Issue by tshort** [tshort/Sims.jl#39]
This was found as part of trying to get Travis to work. Dassl wouldn't install, so `sunsim` was used for everything. After changing `sim` from `dasslsim` to `sunsim`, tests fail with the following message:

```julia
== run all examples
starting sunsim()

[IDAS ERROR]  IDACalcIC
  Newton/Linesearch algorithm failed to converge.

structural change event found at t = 5.012, restarting
ERROR: DimensionMismatch("tried to assign 4 elements to 6 destinations")
 in throw_setindex_mismatch at operators.jl:233
 in setindex! at array.jl:330
 in ##_sim_resid#8644 at /home/tshort/.julia/v0.3/Sims/src/simcreation.jl:255
 in daefun at /home/tshort/.julia/v0.3/Sims/src/sundials.jl:32
while loading /home/tshort/.julia/v0.3/Sims/test/runtests.jl, in expression starting on line 7
```
It may always have been like this. Looks like a problem with structural updates.

**Comment by iraikov** [comment]
Hm, I see now that the breaking_pendulum example has 6 state variables in the initial regime, and switches to 4 variables subsequently. I was assuming that the number of state variables does not change, so I may have broken things. Let me take a look.

**Comment by iraikov** [comment]
I think I refactored sundials to just call IDAReInit when a structural event occurs for efficiency reasons, instead of creating a whole new instance of IDA. However, I see in the IDA documentation that ReInit requires the new problem to be of the same size, which is obviously not the case with the breaking pendulum. I think we can keep this optimization but add a check whether the new problem is a different size and only then create a new instance of IDA. What do you think?

**Comment by tshort** [comment]
That approach sounds good, Ivan.

**Comment by iraikov** [comment]
Ok, I have already made the fix, and will create a pull request when the fix for delay() is merged.

**Comment by tshort** [comment]
Great! Go ahead and merge the delay fix, then.

**Comment by tshort** [comment]
Fixed by #42.
